[quant][graphmode][refactor] Checks for bias and weight

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33650 [not4land] debugging
* #33533 [not4land] Temporary workaround to get quantized ops
* #33927 [quant][graphmode] Add Finalize function that inlines graph and produce quantized ops
* #33895 [quant][graphmode] insert prepack/unpack for aten::linear
* #33854 [quant][graphmode] Quantization pattern for aten::linear
* #33853 [quant][graphmode] Swap quantized functional linear with aten::linear
* #33481 [quant][graphmode] Handling ops doesn't require observation in insertObservers
* #33532 [quant][graphmode] Add quantization pattern for quantized::add_relu
* #33279 [quant][graphmode] Add quantized conv2d-relu fusion pattern
* #32816 [quant][graphmode] Add add_relu pattern in skip values
* #32814 [quant][graphmode] Skip quantizing input and output in matched module
* #33277 [quant][graphmode] Observing input/output values in call site
* #33649 [quant][graphmode][refactor] Factor out getInvokedModule
* #33276 [quant][graphmode][refactor] Move check for weight outside of insertObserverFor
* **#33273 [quant][graphmode][refactor] Checks for bias and weight**
* #33275 [quant][graphmode][refactor] Move values_to_skip check inside valueNeedsToBeQuantized

Summary:
- Move the check for bias to valueNeedsToBeQuantized
- Move TORCH_CHECK inside the functions for checking if a value is bias or weight

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20123595](https://our.internmc.facebook.com/intern/diff/D20123595)

dreiss commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -586,17 +586,28 @@ bool matchArgPattern(
+  if (result) {
+    TORCH_CHECK(
+        v->uses().size() == 1,
+        "We only support conv/linear bias being used by one node.");
```
Maybe change "We only support" to "quantization only supports"?

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Checks for bias and weight"


Summary:
- Move the check for bias to valueNeedsToBeQuantized
- Move TORCH_CHECK inside the functions for checking if a value is bias or weight

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20123595](https://our.internmc.facebook.com/intern/diff/D20123595)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Checks for bias and weight"


Summary:
- Move the check for bias to valueNeedsToBeQuantized
- Move TORCH_CHECK inside the functions for checking if a value is bias or weight

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20123595](https://our.internmc.facebook.com/intern/diff/D20123595)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Checks for bias and weight"


Summary:
- Move the check for bias to valueNeedsToBeQuantized
- Move TORCH_CHECK inside the functions for checking if a value is bias or weight

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D20123595](https://our.internmc.facebook.com/intern/diff/D20123595)

[ghstack-poisoned]
